spyder/widgets/explorer.py

```diff
@@ -123,7 +140,23 @@ def icon(self, icontype_or_qfileinfo):
             if osp.isdir(fname):
                 return ima.icon('DirOpenIcon')
             else:
-                return ima.icon('FileIcon')
+                basename = osp.basename(fname)
+                mime_type, _ = mime.guess_type(basename)
+                icon = ima.icon('FileIcon')
+                if mime_type is not None:
+                    file_type, bin_name = mime_type.split('/')
+                    if file_type == 'text':
+                        icon = ima.icon('TextFileIcon')
+                    elif file_type == 'audio':
+                        icon = ima.icon('AudioFileIcon')

```

Comment:
Although I think it's better to leave it as `AudioFileIcon` and change it in `icon_manager`.